tls/s2n_client_hello.c

```diff
@@ -594,7 +594,13 @@ int s2n_process_client_hello(struct s2n_connection *conn)
     POSIX_CHECKED_MEMCPY(previous_cipher_suite_iana, conn->secure->cipher_suite->iana_value, S2N_TLS_CIPHER_SUITE_LEN);
 
     /* Now choose the ciphers we have certs for. */
-    POSIX_GUARD(s2n_set_cipher_as_tls_server(conn, client_hello->cipher_suites.data, client_hello->cipher_suites.size / 2));
+    if (conn->client_hello_version == S2N_SSLv2) {
+        POSIX_GUARD(s2n_set_cipher_as_sslv2_server(conn, client_hello->cipher_suites.data,
+                client_hello->cipher_suites.size / S2N_SSLv2_CIPHER_SUITE_LEN));
+    } else {
+        POSIX_GUARD(s2n_set_cipher_as_tls_server(conn, client_hello->cipher_suites.data,
+                client_hello->cipher_suites.size / 2));
+    }
```

Comment:
Should we just call s2n_set_cipher_as_sslv2_server from s2n_set_cipher_as_tls_server instead? Probably simpler. s2n_process_client_hello doesn't really need to know that sslv2 does cipher suites differently.